rllib/evaluation/rollout_worker.py

```diff
@@ -1969,8 +1968,21 @@ def _build_policy_map(
                 )
 
             new_policy = self.policy_map[name]
+            restore_state = (
+                policy_states and name in policy_states and policy_states[name]
+            )
+            # Set the state of the newly created policy before syncing filters, etc.
+            if restore_state:
+                new_policy.set_state(policy_states[name])
+
             if merged_conf.enable_connectors:
-                create_connectors_for_policy(new_policy, merged_conf)
+                # Note(jungong) : we should NOT decide whether new connectors
+                # need to be created for this policy by checking whether it
+                # already has connectors or not. We may be restoring a policy
+                # that had 0 connectors configured.
+                if not policy and not restore_state:
+                    # Not restoring an existing policy. Create new connectors.
```

Comment:
Can you modify the comment to this (I think it will be more clear (at least to me)):
```
We should only create new connectors for the policy iff we are creating a new policy from scratch. i.e. We should NOT create new connectors when we already have the policy object created before this function call or have the restoring states from the caller. Also note that we cannot just check the existence of connectors to decide whether we should create connectors because we may be restoring a policy that has 0 connectors configured. 
```

Reply:
done 😄